tests/onboarding_test/test_onboarding_process_mqtt.py

```diff
@@ -0,0 +1,73 @@
+import pytest
+
+from agrirouter.onboarding.onboarding import OnboardingService
+from agrirouter.environments.environments import QAEnvironment
+from agrirouter.onboarding.enums import CertificateTypes, GateWays
+from agrirouter.onboarding.parameters import OnboardParameters
+from agrirouter.onboarding.response import OnboardResponse
+from tests.data import identifier
+from tests.data.applications import CommunicationUnit
+from tests.data.onboard_response_integration_service import OnboardResponseIntegrationService
+from tests.constants import cu_recipient_endpoint_id
+
+
+@pytest.fixture
+def onboarding_process_fixture():
+    def __onboard(uuid: str, _environment, registration_code: str, certification_type_definition: str = "PEM",
+                  gateway_id: str = "2") -> OnboardResponse:
+        onboarding_service = OnboardingService(env=_environment)
+        onboarding_parameters = OnboardParameters(
+            id_=uuid,
+            reg_code=registration_code,
+            certificate_type=certification_type_definition,
+            gateway_id=gateway_id,
+            application_id=CommunicationUnit.application_id,
+            certification_version_id=CommunicationUnit.certification_version_id,
+            time_zone="+01:00"
+        )
+
+        onboard_response = onboarding_service.onboard(onboarding_parameters)
+        assert onboard_response.device_alternate_id != ''
+        assert onboard_response.sensor_alternate_id != ''
+        assert onboard_response.capability_alternate_id != ''
+        assert onboard_response.authentication.certificate != ''
+        assert onboard_response.authentication.secret != ''
+        assert onboard_response.authentication.type != ''
+        assert onboard_response.connection_criteria.commands != ''
+        assert onboard_response.connection_criteria.measures != ''
+
+        return onboard_response
+
+    return __onboard
+
+
+class TestSingleMqttEndpointWithPEMCertificate:
+    _environment = QAEnvironment()
+
+    def test_update_recipient_with_PEM(self, onboarding_process_fixture):
```

Comment:
In an automated scenario, those tests will fail the time, since the registration code will not be valid.

Reply:
Is there an automated way/Function to produce a unique registration code, every time the function is called? This might help with the automated scenario.